### PR TITLE
[Error codes] Stop using reflective 'BaseError.extractContext' to extract error's context

### PR DESCRIPTION
**Pull request by pbatko-da**
See also https://github.com/digital-asset/daml/pull/13197/files#r821684369

### CODE REVIEW

**Review by mziolekda — APPROVED** [review]

File: ledger/error/src/main/scala/com/daml/error/BaseError.scala [hunk 1]
```diff
@@ -101,15 +100,6 @@ object BaseError {
...
 
-  def extractContext[D](obj: D): Map[String, String] =
-    obj.getClass.getDeclaredFields
-      .filterNot(x => ignoreFields.contains(x.getName) || x.getName.startsWith("_"))
-      .map { field =>
-        field.setAccessible(true)
-        (field.getName, field.get(obj).toString)
-      }
-      .toMap
-
```
Comment (tudor-da) [inline ledger/error/src/main/scala/com/daml/error/BaseError.scala:112]: @pbatko-da @mziolekda Did you check with Canton wrt to this change? I've discussed with Ratko a couple of months ago re this, but he did not want to drop the reflective gathering of context.

Comment (tudor-da) [reply ledger/error/src/main/scala/com/daml/error/BaseError.scala:112]: Great. Thanks @rgugliel-da

**Comment by tudor-da** [comment]
Nice one @pbatko-da 👍

File: ledger/error/src/main/scala/com/daml/error/BaseError.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ trait BaseError extends LocationMixin {
     *
     * At the moment, we'll figure them out using reflection.
```
Comment (rgugliel-da) [inline ledger/error/src/main/scala/com/daml/error/BaseError.scala:38]: Remove that part of the comment as part of the next PR?

**Pull request merged**